external/fv3fit/fv3fit/emulation/layers/architecture.py

```diff
@@ -295,17 +308,31 @@ def NoWeightSharingSLP(
 class StandardOutput(tf.keras.layers.Layer):
     """Uses densely-connected layers w/ linear activation"""
 
-    def __init__(self, feature_lengths: Mapping[str, int], *args, **kwargs):
+    def __init__(
+        self,
+        feature_lengths: Mapping[str, int],
+        kernel_regularizer: Optional[Mapping[str, Any]] = None,
```

Comment:
`kernel_regularizer` or `RegularizerConfig` are two new coupling points that could give this class a reason to change. The `dacite.from_dict`call  in  particular is new concern that is liable to change with the configuration frontend. 

A couple of options:
1. pass in the instantiated `self._kernel_regularizer.instance` in directly (this at least addresses the concern of coupling to the front-end configs).
2. Pass in a `dense_layer_factory` as a keyword argument with a default of `tf.keras.layers.Dense`. Build the dense_layer_factory after the configs are built using the kernel regularizer. You can use `dense_layer_factory` above too.

2 is the most testable design and completely removes "regularization" from the list of things these classes need to worry about and it reduces repetition and potentially inconsistent handling of the configs.

Reply:
Thanks that's a good point.

> pass in the instantiated self._kernel_regularizer.instance in directly (this at least addresses the concern of coupling to the front-end configs).

I was worried this would cause issues for serialization (in particular adding the keras regularizer object to the `get_config` call) but that seems to not be the case.

> Pass in a dense_layer_factory as a keyword argument with a default of tf.keras.layers.Dense. Build the dense_layer_factory after the configs are built using the kernel regularizer. You can use dense_layer_factory above too.

Agreed this would clean things up. So I should also  remove `width` and `activation` from the MLPBlock args, since these would be handled by the dense_layer_factor?

Reply:
> Agreed this would clean things up. So I should also remove width and activation from the MLPBlock args, since these would be handled by the dense_layer_factor?

Hmm.. that feels a bit too far for me but it's a subjective call. Up to you.